src/contexts/NetworkContext.tsx

```diff
@@ -36,58 +54,43 @@ const NetworkContextProvider = ({ children }: NetworkContextProps) => {
   const [api, setApi] = useState<ApiType>()
 
   const [assetInfo, setAssetInfo] = useState<AssetType>({} as AssetType)
-  const [network, setNetwork] = useState<NetworkProps>('polkadot')
+  const [network, setNetwork] = useState<SupportedNetworkNames>('polkadot')
 
   useEffect(() => {
-    let cl: PolkadotClient
-    let typedApi: ApiType
-
-    switch (network) {
-      case 'polkadot':
-        {
-          const { assetInfo, wsEndpoint } = getChainInformation('polkadot')
-          setAssetInfo(assetInfo)
-          setIsLight(false)
-          cl = createClient(getWsProvider(wsEndpoint))
-          typedApi = cl.getTypedApi(dot)
-        }
-        break
-      case 'polkadot-lc': {
-        const { assetInfo } = getChainInformation('polkadot')
-        setAssetInfo(assetInfo)
-        setIsLight(true)
-        const smoldot = startFromWorker(new SmWorker())
-        const dotRelayChain = import('polkadot-api/chains/polkadot').then(
+    let client: PolkadotClient
+
+    if (network === 'polkadot-lc' || network === 'kusama-lc') {
+      const relay = network === 'polkadot-lc' ? 'polkadot' : 'kusama'
+
+      const { assetInfo } = getChainInformation(relay)
+      setAssetInfo(assetInfo)
+      setIsLight(true)
+      const smoldot = startFromWorker(new SmWorker())
+      let relayChain: Promise<unknown>
+      if (relay === 'polkadot') {
+        relayChain = import('polkadot-api/chains/polkadot').then(
           ({ chainSpec }) => smoldot.addChain({ chainSpec }),
         )
-        cl = createClient(getSmProvider(dotRelayChain))
-        typedApi = cl.getTypedApi(dot)
-        break
-      }
-      case 'kusama':
-        {
-          const { assetInfo, wsEndpoint } = getChainInformation('kusama')
-          setAssetInfo(assetInfo)
-          setIsLight(false)
-
-          cl = createClient(getWsProvider(wsEndpoint))
-          typedApi = cl.getTypedApi(ksm)
-        }
-        break
-      case 'kusama-lc': {
-        const { assetInfo } = getChainInformation('kusama')
-        setAssetInfo(assetInfo)
-        setIsLight(true)
-        const smoldot = startFromWorker(new SmWorker())
-        const ksmRelayChain = import('polkadot-api/chains/ksmcc3').then(
+      } else {
+        relayChain = import('polkadot-api/chains/ksmcc3').then(
           ({ chainSpec }) => smoldot.addChain({ chainSpec }),
         )
-        cl = createClient(getSmProvider(ksmRelayChain))
-        typedApi = cl.getTypedApi(ksm)
-        break
       }
+      //@ts-expect-error the Chain type isn't exported
+      //it comes from 'smoldot' that we don't import
+      client = createClient(getSmProvider(relayChain as unknown))
```

Comment:
Smoldot can be imported from polkadot-api as well so you can add at the top:
```ts
import type { Chain } from 'polkadot-api/smoldot'
```
and then:
```suggestion
      client = createClient(getSmProvider(relayChain as Chain))
```

Reply:
oh nice, somehow my IDE didn't see it